test /var/fworch/lock/testhost_log.lock being used by another process

here on a fresh install debian 11 during first upgrade attempt (develop 6.4.3 to 6.4.4 tpurschke branch)

failed again during upgrade retry

But dir does not contain file (after failed test):
```
fworch@debian11:~/importer$ ls -l /var/fworch/lock
total 0
-rw-r--r-- 1 fworch fworch 0 Jun 19 08:14 FWO.Middleware.Server_log.lock
-rw-r--r-- 1 fworch fworch 0 Jun 19 08:14 FWO.Ui_log.lock
fworch@debian11:~/importer$ 
```
**Upgrade Log:**
```
TASK [test : csharp tests] **********************************************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => changed=true 
  cmd:
  - dotnet
  - test
  delta: '0:00:09.148347'
  end: '2023-06-19 08:39:58.627451'
  msg: non-zero return code
  rc: 1
  start: '2023-06-19 08:39:49.479104'
  stderr: ''
  stderr_lines: <omitted>
  stdout: |2-
      Determining projects to restore...
      All projects are up-to-date for restore.
      FWO.Logging -> /usr/local/fworch/lib/files/FWO.Logging/bin/Debug/net6.0/FWO.Logging.dll
      FWO.Mail -> /usr/local/fworch/lib/files/FWO.Mail/bin/Debug/net6.0/FWO.Mail.dll
      FWO.Config.File -> /usr/local/fworch/lib/files/FWO.Config.File/bin/Debug/net6.0/FWO.Config.File.dll
      FWO.Middleware -> /usr/local/fworch/lib/files/FWO.Middleware/bin/Debug/net6.0/FWO.Middleware.dll
    /usr/local/fworch/lib/files/FWO.Api.Client/ApiSubscription.cs(27,16): warning CS8618: Non-nullable field 'subscriptionStream' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/usr/local/fworch/lib/files/FWO.Api.Client/FWO.Api.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Api.Client/ApiSubscription.cs(27,16): warning CS8618: Non-nullable field 'subscription' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/usr/local/fworch/lib/files/FWO.Api.Client/FWO.Api.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Api.Client/ApiSubscription.cs(61,21): warning CS0162: Unreachable code detected [/usr/local/fworch/lib/files/FWO.Api.Client/FWO.Api.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Api.Client/GraphQlApiConnection.cs(50,16): warning CS8618: Non-nullable property 'ApiServerUri' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [/usr/local/fworch/lib/files/FWO.Api.Client/FWO.Api.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Api.Client/GraphQlApiConnection.cs(50,16): warning CS8618: Non-nullable field 'graphQlClient' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/usr/local/fworch/lib/files/FWO.Api.Client/FWO.Api.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Api.Client/GraphQlApiConnection.cs(56,16): warning CS8618: Non-nullable property 'ApiServerUri' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [/usr/local/fworch/lib/files/FWO.Api.Client/FWO.Api.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Api.Client/GraphQlApiConnection.cs(56,16): warning CS8618: Non-nullable field 'graphQlClient' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/usr/local/fworch/lib/files/FWO.Api.Client/FWO.Api.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Api.Client/GraphQlApiConnection.cs(119,25): warning CS0162: Unreachable code detected [/usr/local/fworch/lib/files/FWO.Api.Client/FWO.Api.Client.csproj]
      FWO.Api.Client -> /usr/local/fworch/lib/files/FWO.Api.Client/bin/Debug/net6.0/FWO.Api.Client.dll
    /usr/local/fworch/lib/files/FWO.Config.Api/Config.cs(32,19): warning CS8618: Non-nullable field 'apiConnection' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/usr/local/fworch/lib/files/FWO.Config.Api/FWO.Config.Api.csproj]
    /usr/local/fworch/lib/files/FWO.Config.Api/Config.cs(32,19): warning CS8618: Non-nullable property 'RawConfigItems' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [/usr/local/fworch/lib/files/FWO.Config.Api/FWO.Config.Api.csproj]
    /usr/local/fworch/lib/files/FWO.Config.Api/Config.cs(34,19): warning CS8618: Non-nullable field 'apiConnection' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/usr/local/fworch/lib/files/FWO.Config.Api/FWO.Config.Api.csproj]
    /usr/local/fworch/lib/files/FWO.Config.Api/Config.cs(34,19): warning CS8618: Non-nullable property 'RawConfigItems' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [/usr/local/fworch/lib/files/FWO.Config.Api/FWO.Config.Api.csproj]
    /usr/local/fworch/lib/files/FWO.Config.Api/UserConfig.cs(52,19): warning CS8618: Non-nullable field 'globalConfig' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/usr/local/fworch/lib/files/FWO.Config.Api/FWO.Config.Api.csproj]
    /usr/local/fworch/lib/files/FWO.Config.Api/UserConfig.cs(52,19): warning CS8618: Non-nullable property 'Translate' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [/usr/local/fworch/lib/files/FWO.Config.Api/FWO.Config.Api.csproj]
    /usr/local/fworch/lib/files/FWO.Config.Api/UserConfig.cs(52,19): warning CS8618: Non-nullable property 'User' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [/usr/local/fworch/lib/files/FWO.Config.Api/FWO.Config.Api.csproj]
    /usr/local/fworch/lib/files/FWO.Config.Api/UserConfig.cs(90,34): warning CS8625: Cannot convert null literal to non-nullable reference type. [/usr/local/fworch/lib/files/FWO.Config.Api/FWO.Config.Api.csproj]
      FWO.Middleware.Client -> /usr/local/fworch/lib/files/FWO.Middleware.Client/bin/Debug/net6.0/FWO.Middleware.Client.dll
    /usr/local/fworch/lib/files/FWO.Report.Filter/Ast/AstNodeFilterDateTimeRange.cs(26,49): warning CS8629: Nullable value type may be null. [/usr/local/fworch/lib/files/FWO.Report.Filter/FWO.Report.Filter.csproj]
    /usr/local/fworch/lib/files/FWO.Report.Filter/Ast/AstNodeFilterDateTimeRange.cs(11,23): warning CS8618: Non-nullable field 'semanticValue' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/usr/local/fworch/lib/files/FWO.Report.Filter/FWO.Report.Filter.csproj]
      FWO.Report.Filter -> /usr/local/fworch/lib/files/FWO.Report.Filter/bin/Debug/net6.0/FWO.Report.Filter.dll
      FWO.Config.Api -> /usr/local/fworch/lib/files/FWO.Config.Api/bin/Debug/net6.0/FWO.Config.Api.dll
      FWO.DeviceAutoDiscovery -> /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/bin/Debug/net6.0/FWO.DeviceAutoDiscovery.dll
      FWO.Report -> /usr/local/fworch/lib/files/FWO.Report/bin/Debug/net6.0/FWO.Report.dll
    /usr/local/fworch/test/csharp/FWO.Test/ApiTest.cs(19,16): warning CS8618: Non-nullable field 'apiConnection' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]
    /usr/local/fworch/test/csharp/FWO.Test/ApiTest.cs(51,27): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]
    /usr/local/fworch/test/csharp/FWO.Test/ApiConfigTest.cs(30,27): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]
    /usr/local/fworch/test/csharp/FWO.Test/FakeLocalTimeZone.cs(12,33): warning CS8602: Dereference of a possibly null reference. [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]
    /usr/local/fworch/test/csharp/FWO.Test/FakeLocalTimeZone.cs(12,33): warning CS8600: Converting null literal or possible null value to non-nullable type. [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]
    /usr/local/fworch/test/csharp/FWO.Test/FakeLocalTimeZone.cs(14,25): warning CS8602: Dereference of a possibly null reference. [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]
    /usr/local/fworch/test/csharp/FWO.Test/FakeLocalTimeZone.cs(15,13): warning CS8602: Dereference of a possibly null reference. [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]
    /usr/local/fworch/test/csharp/FWO.Test/ApiTest.cs(17,23): warning CS0169: The field 'ApiTest.apiConnection' is never used [/usr/local/fworch/test/csharp/FWO.Test/FWO.Test.csproj]
      FWO.Test -> /usr/local/fworch/test/csharp/FWO.Test/bin/Debug/net6.0/FWO.Test.dll
    Test run for /usr/local/fworch/test/csharp/FWO.Test/bin/Debug/net6.0/FWO.Test.dll (.NETCoreApp,Version=v6.0)
    Microsoft (R) Test Execution Command Line Tool Version 17.3.3 (x64)
    Copyright (c) Microsoft Corporation.  All rights reserved.
  
    Starting test execution, please wait...
    A total of 1 test files matched the specified pattern.
    2023-06-19T08:39:57+02:00 Info - Startup (LockTest.cs in line 22), Starting Lock Tests...
  
    2023-06-19T08:39:57+02:00 Debug - Filter (Compiler.cs in line 23), Input: "disabled == true", Report Type: "$1", Device Filter: ""
  
      Failed LogLockUi [38 ms]
      Error Message:
       System.IO.IOException : The process cannot access the file '/var/fworch/lock/testhost_log.lock' because it is being used by another process.
      Stack Trace:
         at Microsoft.Win32.SafeHandles.SafeFileHandle.Init(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
       at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
       at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
       at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
       at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
       at System.IO.StreamWriter..ctor(String path)
       at FWO.Test.LockTest.LogLockUi() in /usr/local/fworch/test/csharp/FWO.Test/LockTest.cs:line 39
       at NUnit.Framework.Internal.TaskAwaitAdapter.GenericAdapter`1.GetResult()
       at NUnit.Framework.Internal.AsyncToSyncAdapter.Await(Func`1 invoke)
       at NUnit.Framework.Internal.Commands.TestMethodCommand.RunTestMethod(TestExecutionContext context)
       at NUnit.Framework.Internal.Commands.TestMethodCommand.Execute(TestExecutionContext context)
       at NUnit.Framework.Internal.Commands.BeforeAndAfterTestCommand.<>c__DisplayClass1_0.<Execute>b__0()
       at NUnit.Framework.Internal.Commands.DelegatingTestCommand.RunTestMethodInThreadAbortSafeZone(TestExecutionContext context, Action action)
  
      Standard Output Messages:
     2023-06-19T08:39:57+02:00 Info - Startup (LockTest.cs in line 22), Starting Lock Tests...
  
  
    2023-06-19T08:39:57+02:00 Debug - Filter (Compiler.cs in line 23), Input: "", Report Type: "$1", Device Filter: ""
  
    2023-06-19T08:39:57+02:00 Debug - Filter (Compiler.cs in line 23), Input: "(text==cactus)", Report Type: "$1", Device Filter: ""
  
    2023-06-19T08:39:57+02:00 Debug - Filter (Compiler.cs in line 23), Input: "src=a&(dst=c)", Report Type: "$1", Device Filter: ""
  
    2023-06-19T08:39:57+02:00 Debug - Filter (Compiler.cs in line 23), Input: "(gateway="checkpoint_demo" or gateway = "fortigate_demo") & dst ==", Report Type: "$1", Device Filter: ""
  
    2023-06-19T08:39:57+02:00 Debug - Filter (Compiler.cs in line 23), Input: "(text!=cactus)", Report Type: "$1", Device Filter: ""
  
    2023-06-19T08:39:57+02:00 Debug - Filter (Compiler.cs in line 23), Input: "teststring", Report Type: "$1", Device Filter: ""
  
    2023-06-19T08:39:57+02:00 Debug - Filter (Compiler.cs in line 23), Input: "(gateway = "checkpoint_demo" or gateway = "fortigate_demo") & dst == IsoAAADray.local", Report Type: "$1", Device Filter: ""
  
    2023-06-19T08:39:57+02:00 Debug - Filter (Compiler.cs in line 23), Input: "((src=hi) & (dst=test)) | (src = a)", Report Type: "$1", Device Filter: ""
  
    2023-06-19T08:39:57+02:00 Debug - Filter (Compiler.cs in line 23), Input: "(src=cactus or dst=cactus or svc=smtps)", Report Type: "$1", Device Filter: ""
  
    2023-06-19T08:39:57+02:00 Debug - Filter (Compiler.cs in line 23), Input: "
  
  
    ", Report Type: "$1", Device Filter: ""
  
    2023-06-19T08:39:57+02:00 Debug - Key extraction (KeyImporter.cs in line 76), Key was succesfully extracted.
    2023-06-19T08:39:57+02:00 Debug - Key extraction (KeyImporter.cs in line 76), Key was succesfully extracted.
    2023-06-19T08:39:57+02:00 Error - Key extraction (KeyImporter.cs in line 72), Error while trying to read key from file.
     ---
    Exception thrown:
     ArgumentOutOfRangeException
    Message:
     Non-negative number required. (Parameter 'count')
    Stack Trace:
     at System.Collections.Generic.List`1.GetRange(Int32 index, Int32 count)
       at FWO.Config.File.KeyImporter.ExtractKeyFromPemAsString(String rawKey) in /usr/local/fworch/lib/files/FWO.Config.File/KeyImporter.cs:line 68
    2023-06-19T08:39:57+02:00 Debug - Config value (ConfigFile.cs in line 179), Config value could not be loaded. Error: Non-negative number required. (Parameter 'count')
    2023-06-19T08:39:57+02:00 Error - Key extraction (KeyImporter.cs in line 72), Error while trying to read key from file.
     ---
    Exception thrown:
     ArgumentOutOfRangeException
    Message:
     Non-negative number required. (Parameter 'count')
    Stack Trace:
     at System.Collections.Generic.List`1.GetRange(Int32 index, Int32 count)
       at FWO.Config.File.KeyImporter.ExtractKeyFromPemAsString(String rawKey) in /usr/local/fworch/lib/files/FWO.Config.File/KeyImporter.cs:line 68
    2023-06-19T08:39:57+02:00 Debug - Config value (ConfigFile.cs in line 179), Config value could not be loaded. Error: Non-negative number required. (Parameter 'count')
  
    2023-06-19T08:39:57+02:00 Debug - Key extraction (KeyImporter.cs in line 76), Key was succesfully extracted.
    2023-06-19T08:39:57+02:00 Error - Key extraction (KeyImporter.cs in line 72), Error while trying to read key from file.
     ---
    Exception thrown:
     ArgumentOutOfRangeException
    Message:
     Non-negative number required. (Parameter 'count')
    Stack Trace:
     at System.Collections.Generic.List`1.GetRange(Int32 index, Int32 count)
       at FWO.Config.File.KeyImporter.ExtractKeyFromPemAsString(String rawKey) in /usr/local/fworch/lib/files/FWO.Config.File/KeyImporter.cs:line 68
    2023-06-19T08:39:57+02:00 Debug - Config value (ConfigFile.cs in line 179), Config value could not be loaded. Error: Non-negative number required. (Parameter 'count')
    2023-06-19T08:39:57+02:00 Debug - Key extraction (KeyImporter.cs in line 76), Key was succesfully extracted.
  
    2023-06-19T08:39:57+02:00 Error - Key extraction (KeyImporter.cs in line 72), Error while trying to read key from file.
     ---
    Exception thrown:
     ArgumentOutOfRangeException
    Message:
     Non-negative number required. (Parameter 'count')
    Stack Trace:
     at System.Collections.Generic.List`1.GetRange(Int32 index, Int32 count)
       at FWO.Config.File.KeyImporter.ExtractKeyFromPemAsString(String rawKey) in /usr/local/fworch/lib/files/FWO.Config.File/KeyImporter.cs:line 68
    2023-06-19T08:39:57+02:00 Debug - Config value (ConfigFile.cs in line 179), Config value could not be loaded. Error: Non-negative number required. (Parameter 'count')
    2023-06-19T08:39:57+02:00 Debug - Key extraction (KeyImporter.cs in line 76), Key was succesfully extracted.
    2023-06-19T08:39:57+02:00 Debug - Key extraction (KeyImporter.cs in line 76), Key was succesfully extracted.
  
    2023-06-19T08:39:57+02:00 Debug - Key extraction (KeyImporter.cs in line 76), Key was succesfully extracted.
    2023-06-19T08:39:57+02:00 Error - Extract Key (KeyImporter.cs in line 48), unexpected error while trying to extract rsakey from PEM formatted key
    lV4dPfPYm+Xw+qiHJT7FCUG3Vz5eaYaxYwE8l0IcQ56p0VF2pwfjsHkhGJIlMzSg
    JodQLRlUP8/Lazz7YTSqytGef7XE88JvBzOsB8z0hHyq1BiFbgJlxwovlGdJasla
    ubvqQPOqeHqYBtec5DdQyxBh+nmRd13LqPEPCwKBgHiJjajCZZCzy5tmaOYL52Pe
    4MEORmMWEjBAOYEQxsc+9ie1yw7v/gzzYsjev1LezjP6BLrUelbpQLoGkY5h4rDG
    9d4xEXtjqwWPQMonTzVWcO6PhN9WGdhlE2kKRbJHV+Ylk2JfL4G4HXpHT0JXsQSj
    +rUCgYARFC0N0BWTtB0ZmhhWXL4zWeCmlUV/96EkKD3Qrkj2AFK8Sj3O96U5TXR2
    5+uY7nOytVkLEz5cz9+blzXEQxf+MtZ0wC7g5zhAC2Yvfm9YSYh9pNlQgFOEJfKU
    QdJmDaQOJ2fLoK7HjP60Lq4qcUdW79BdPJS4njwLp/FmsebraQ==ÖQWRSO
    -----END KEY-----.
     ---
    Exception thrown:
     FormatException
    Message:
     The input is not a valid Base-64 string as it contains a non-base 64 character, more than two padding characters, or an illegal character among the padding characters.
    Stack Trace:
     at System.Convert.FromBase64CharPtr(Char* inputPtr, Int32 inputLength)
       at System.Convert.FromBase64String(String s)
       at FWO.Config.File.KeyImporter.ExtractKeyFromPem(String rawKey, Boolean isPrivateKey) in /usr/local/fworch/lib/files/FWO.Config.File/KeyImporter.cs:line 18
    2023-06-19T08:39:57+02:00 Debug - Config value (ConfigFile.cs in line 179), Config value could not be loaded. Error: The input is not a valid Base-64 string as it contains a non-base 64 character, more than two padding characters, or an illegal character among the padding characters.
    2023-06-19T08:39:57+02:00 Error - Key extraction (KeyImporter.cs in line 72), Error while trying to read key from file.
     ---
    Exception thrown:
     ArgumentOutOfRangeException
    Message:
     Non-negative number required. (Parameter 'count')
    Stack Trace:
     at System.Collections.Generic.List`1.GetRange(Int32 index, Int32 count)
       at FWO.Config.File.KeyImporter.ExtractKeyFromPemAsString(String rawKey) in /usr/local/fworch/lib/files/FWO.Config.File/KeyImporter.cs:line 68
    2023-06-19T08:39:57+02:00 Debug - Config value (ConfigFile.cs in line 179), Config value could not be loaded. Error: Non-negative number required. (Parameter 'count')
    2023-06-19T08:39:57+02:00 Error - Config value read (ConfigFile.cs in line 165), A necessary config value could not be found.
     ---
    Stack Trace:
        at System.Environment.get_StackTrace()
       at FWO.Logging.Log.WriteError(String Title, String Text, Boolean LogStackTrace, String callerName, String callerFile, Int32 callerLineNumber) in /usr/local/fworch/lib/files/FWO.Logging/Log.cs:line 138
       at FWO.Config.File.ConfigFile.CriticalConfigValueLoaded[ConfigValueType](ConfigValueType configValue) in /usr/local/fworch/lib/files/FWO.Config.File/ConfigFile.cs:line 165
       at FWO.Config.File.ConfigFile.get_JwtPrivateKey() in /usr/local/fworch/lib/files/FWO.Config.File/ConfigFile.cs:line 44
       at FWO.Test.ConfigFileTest.<>c.<IncorrectPrivateKey>b__16_0() in /usr/local/fworch/test/csharp/FWO.Test/ConfigFileTest.cs:line 159
       at NUnit.Framework.Assert.Throws(IResolveConstraint expression, TestDelegate code, String message, Object[] args)
       at NUnit.Framework.Assert.Throws(IResolveConstraint expression, TestDelegate code)
       at NUnit.Framework.Assert.Catch(Type expectedExceptionType, TestDelegate code)
       at FWO.Test.ConfigFileTest.IncorrectPrivateKey() in /usr/local/fworch/test/csharp/FWO.Test/ConfigFileTest.cs:line 159
       at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
       at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
       at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
       at NUnit.Framework.Internal.Reflect.InvokeMethod(MethodInfo method, Object fixture, Object[] args)
       at NUnit.Framework.Internal.MethodWrapper.Invoke(Object fixture, Object[] args)
       at NUnit.Framework.Internal.Commands.TestMethodCommand.InvokeTestMethod(TestExecutionContext context)
       at NUnit.Framework.Internal.Commands.TestMethodCommand.RunTestMethod(TestExecutionContext context)
       at NUnit.Framework.Internal.Commands.TestMethodCommand.Execute(TestExecutionContext context)
       at NUnit.Framework.Internal.Execution.SimpleWorkItem.<>c__DisplayClass4_0.<PerformWork>b__0()
       at NUnit.Framework.Internal.ContextUtils.<>c__DisplayClass1_0`1.<DoIsolated>b__0(Object _)
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
       at NUnit.Framework.Internal.ContextUtils.DoIsolated(ContextCallback callback, Object state)
       at NUnit.Framework.Internal.ContextUtils.DoIsolated[T](Func`1 func)
       at NUnit.Framework.Internal.Execution.SimpleWorkItem.PerformWork()
       at NUnit.Framework.Internal.Execution.WorkItem.RunOnCurrentThread()
       at NUnit.Framework.Internal.Execution.WorkItem.Execute()
       at NUnit.Framework.Internal.Execution.ParallelWorkItemDispatcher.Dispatch(WorkItem work, ParallelExecutionStrategy strategy)
       at NUnit.Framework.Internal.Execution.ParallelWorkItemDispatcher.Dispatch(WorkItem work)
       at NUnit.Framework.Internal.Execution.CompositeWorkItem.RunChildren()
       at NUnit.Framework.Internal.Execution.CompositeWorkItem.PerformWork()
       at NUnit.Framework.Internal.Execution.WorkItem.RunOnCurrentThread()
       at NUnit.Framework.Internal.Execution.WorkItem.Execute()
       at NUnit.Framework.Internal.Execution.TestWorker.TestWorkerThreadProc()
       at System.Threading.Thread.StartHelper.Callback(Object state)
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.Thread.StartCallback()
  
    2023-06-19T08:39:57+02:00 Error - Key extraction (KeyImporter.cs in line 72), Error while trying to read key from file.
     ---
    Exception thrown:
     ArgumentOutOfRangeException
    Message:
     Non-negative number required. (Parameter 'count')
    Stack Trace:
     at System.Collections.Generic.List`1.GetRange(Int32 index, Int32 count)
       at FWO.Config.File.KeyImporter.ExtractKeyFromPemAsString(String rawKey) in /usr/local/fworch/lib/files/FWO.Config.File/KeyImporter.cs:line 68
    2023-06-19T08:39:57+02:00 Debug - Config value (ConfigFile.cs in line 179), Config value could not be loaded. Error: Non-negative number required. (Parameter 'count')
    2023-06-19T08:39:57+02:00 Debug - Key extraction (KeyImporter.cs in line 76), Key was succesfully extracted.
    2023-06-19T08:39:57+02:00 Error - Extract Key (KeyImporter.cs in line 48), unexpected error while trying to extract rsakey from PEM formatted key ---- BEGIN
    HALOOOOOOOOOOOOÖÖÖÖÖÖÖÖÖÖÄ
    aMYhfZZP8YaYhwhjEkH345kAp5+YVthfBuPEeksCUplfMYuZhdByolHm7WzVLXFe
    yC4oOfBfWUUenF7nHr4TW86k9pT1LZTcukPI00BpZzsiz7+1JZ0YoOIIA1u4672B
    z2cAR6HkNFB63sh2qZwtC0utP3i3yXlDSxD8lQ7A7NYlifRszw==
    2 PUBLIC KEY ----.
     ---
    Exception thrown:
     FormatException
    Message:
     The input is not a valid Base-64 string as it contains a non-base 64 character, more than two padding characters, or an illegal character among the padding characters.
    Stack Trace:
     at System.Convert.FromBase64CharPtr(Char* inputPtr, Int32 inputLength)
       at System.Convert.FromBase64String(String s)
       at FWO.Config.File.KeyImporter.ExtractKeyFromPem(String rawKey, Boolean isPrivateKey) in /usr/local/fworch/lib/files/FWO.Config.File/KeyImporter.cs:line 18
    2023-06-19T08:39:57+02:00 Debug - Config value (ConfigFile.cs in line 179), Config value could not be loaded. Error: The input is not a valid Base-64 string as it contains a non-base 64 character, more than two padding characters, or an illegal character among the padding characters.
    2023-06-19T08:39:57+02:00 Error - Config value read (ConfigFile.cs in line 165), A necessary config value could not be found.
     ---
    Stack Trace:
        at System.Environment.get_StackTrace()
       at FWO.Logging.Log.WriteError(String Title, String Text, Boolean LogStackTrace, String callerName, String callerFile, Int32 callerLineNumber) in /usr/local/fworch/lib/files/FWO.Logging/Log.cs:line 138
       at FWO.Config.File.ConfigFile.CriticalConfigValueLoaded[ConfigValueType](ConfigValueType configValue) in /usr/local/fworch/lib/files/FWO.Config.File/ConfigFile.cs:line 165
       at FWO.Config.File.ConfigFile.get_JwtPublicKey() in /usr/local/fworch/lib/files/FWO.Config.File/ConfigFile.cs:line 54
       at FWO.Test.ConfigFileTest.<>c.<IncorrectPublicKey>b__15_0() in /usr/local/fworch/test/csharp/FWO.Test/ConfigFileTest.cs:line 152
       at NUnit.Framework.Assert.Throws(IResolveConstraint expression, TestDelegate code, String message, Object[] args)
       at NUnit.Framework.Assert.Throws(IResolveConstraint expression, TestDelegate code)
       at NUnit.Framework.Assert.Catch(Type expectedExceptionType, TestDelegate code)
       at FWO.Test.ConfigFileTest.IncorrectPublicKey() in /usr/local/fworch/test/csharp/FWO.Test/ConfigFileTest.cs:line 152
       at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
       at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
       at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
       at NUnit.Framework.Internal.Reflect.InvokeMethod(MethodInfo method, Object fixture, Object[] args)
       at NUnit.Framework.Internal.MethodWrapper.Invoke(Object fixture, Object[] args)
       at NUnit.Framework.Internal.Commands.TestMethodCommand.InvokeTestMethod(TestExecutionContext context)
       at NUnit.Framework.Internal.Commands.TestMethodCommand.RunTestMethod(TestExecutionContext context)
       at NUnit.Framework.Internal.Commands.TestMethodCommand.Execute(TestExecutionContext context)
       at NUnit.Framework.Internal.Execution.SimpleWorkItem.<>c__DisplayClass4_0.<PerformWork>b__0()
       at NUnit.Framework.Internal.ContextUtils.<>c__DisplayClass1_0`1.<DoIsolated>b__0(Object _)
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
       at NUnit.Framework.Internal.ContextUtils.DoIsolated(ContextCallback callback, Object state)
       at NUnit.Framework.Internal.ContextUtils.DoIsolated[T](Func`1 func)
       at NUnit.Framework.Internal.Execution.SimpleWorkItem.PerformWork()
       at NUnit.Framework.Internal.Execution.WorkItem.RunOnCurrentThread()
       at NUnit.Framework.Internal.Execution.WorkItem.Execute()
       at NUnit.Framework.Internal.Execution.ParallelWorkItemDispatcher.Dispatch(WorkItem work, ParallelExecutionStrategy strategy)
       at NUnit.Framework.Internal.Execution.ParallelWorkItemDispatcher.Dispatch(WorkItem work)
       at NUnit.Framework.Internal.Execution.CompositeWorkItem.RunChildren()
       at NUnit.Framework.Internal.Execution.CompositeWorkItem.PerformWork()
       at NUnit.Framework.Internal.Execution.WorkItem.RunOnCurrentThread()
       at NUnit.Framework.Internal.Execution.WorkItem.Execute()
       at NUnit.Framework.Internal.Execution.TestWorker.TestWorkerThreadProc()
       at System.Threading.Thread.StartHelper.Callback(Object state)
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.Thread.StartCallback()
  
    2023-06-19T08:39:57+02:00 Error - Config file read (ConfigFile.cs in line 153), Config file could not be found.
     ---
    Exception thrown:
     JsonException
    Message:
     '=' is invalid after a property name. Expected a ':'. Path: $ | LineNumber: 1 | BytePositionInLine: 31.
    Stack Trace:
     at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& state, JsonReaderException ex)
       at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
       at System.Text.Json.JsonSerializer.ReadFromSpan[TValue](ReadOnlySpan`1 utf8Json, JsonTypeInfo jsonTypeInfo, Nullable`1 actualByteCount)
       at System.Text.Json.JsonSerializer.ReadFromSpan[TValue](ReadOnlySpan`1 json, JsonTypeInfo jsonTypeInfo)
       at System.Text.Json.JsonSerializer.Deserialize[TValue](String json, JsonSerializerOptions options)
       at FWO.Config.File.ConfigFile.Read(String configFilePath, String privateKeyFilePath, String publicKeyFilePath) in /usr/local/fworch/lib/files/FWO.Config.File/ConfigFile.cs:line 129
  
    2023-06-19T08:39:57+02:00 Error - Key extraction (KeyImporter.cs in line 72), Error while trying to read key from file.
     ---
    Exception thrown:
     ArgumentOutOfRangeException
    Message:
     Non-negative number required. (Parameter 'count')
    Stack Trace:
     at System.Collections.Generic.List`1.GetRange(Int32 index, Int32 count)
       at FWO.Config.File.KeyImporter.ExtractKeyFromPemAsString(String rawKey) in /usr/local/fworch/lib/files/FWO.Config.File/KeyImporter.cs:line 68
    2023-06-19T08:39:57+02:00 Debug - Config value (ConfigFile.cs in line 179), Config value could not be loaded. Error: Non-negative number required. (Parameter 'count')
    2023-06-19T08:39:57+02:00 Error - Key extraction (KeyImporter.cs in line 72), Error while trying to read key from file.
     ---
    Exception thrown:
     ArgumentOutOfRangeException
    Message:
     Non-negative number required. (Parameter 'count')
    Stack Trace:
     at System.Collections.Generic.List`1.GetRange(Int32 index, Int32 count)
       at FWO.Config.File.KeyImporter.ExtractKeyFromPemAsString(String rawKey) in /usr/local/fworch/lib/files/FWO.Config.File/KeyImporter.cs:line 68
    2023-06-19T08:39:57+02:00 Debug - Config value (ConfigFile.cs in line 179), Config value could not be loaded. Error: Non-negative number required. (Parameter 'count')
    2023-06-19T08:39:57+02:00 Debug - Config value (ConfigFile.cs in line 179), Config value could not be loaded. Error: The given key 'middleware_uri' was not present in the dictionary.
    2023-06-19T08:39:57+02:00 Debug - Config value (ConfigFile.cs in line 179), Config value could not be loaded. Error: The given key 'api_uri' was not present in the dictionary.
    2023-06-19T08:39:57+02:00 Error - Config value read (ConfigFile.cs in line 165), A necessary config value could not be found.
     ---
    Stack Trace:
        at System.Environment.get_StackTrace()
       at FWO.Logging.Log.WriteError(String Title, String Text, Boolean LogStackTrace, String callerName, String callerFile, Int32 callerLineNumber) in /usr/local/fworch/lib/files/FWO.Logging/Log.cs:line 138
       at FWO.Config.File.ConfigFile.CriticalConfigValueLoaded[ConfigValueType](ConfigValueType configValue) in /usr/local/fworch/lib/files/FWO.Config.File/ConfigFile.cs:line 165
       at FWO.Config.File.ConfigFile.get_MiddlewareServerUri() in /usr/local/fworch/lib/files/FWO.Config.File/ConfigFile.cs:line 84
       at FWO.Test.ConfigFileTest.<>c.<MissingValueConfigFile>b__12_0() in /usr/local/fworch/test/csharp/FWO.Test/ConfigFileTest.cs:line 129
       at NUnit.Framework.Assert.Throws(IResolveConstraint expression, TestDelegate code, String message, Object[] args)
       at NUnit.Framework.Assert.Throws(IResolveConstraint expression, TestDelegate code)
       at NUnit.Framework.Assert.Catch(Type expectedExceptionType, TestDelegate code)
       at FWO.Test.ConfigFileTest.MissingValueConfigFile() in /usr/local/fworch/test/csharp/FWO.Test/ConfigFileTest.cs:line 129
       at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
       at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
       at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
       at NUnit.Framework.Internal.Reflect.InvokeMethod(MethodInfo method, Object fixture, Object[] args)
       at NUnit.Framework.Internal.MethodWrapper.Invoke(Object fixture, Object[] args)
       at NUnit.Framework.Internal.Commands.TestMethodCommand.InvokeTestMethod(TestExecutionContext context)
       at NUnit.Framework.Internal.Commands.TestMethodCommand.RunTestMethod(TestExecutionContext context)
       at NUnit.Framework.Internal.Commands.TestMethodCommand.Execute(TestExecutionContext context)
       at NUnit.Framework.Internal.Execution.SimpleWorkItem.<>c__DisplayClass4_0.<PerformWork>b__0()
       at NUnit.Framework.Internal.ContextUtils.<>c__DisplayClass1_0`1.<DoIsolated>b__0(Object _)
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
       at NUnit.Framework.Internal.ContextUtils.DoIsolated(ContextCallback callback, Object state)
       at NUnit.Framework.Internal.ContextUtils.DoIsolated[T](Func`1 func)
       at NUnit.Framework.Internal.Execution.SimpleWorkItem.PerformWork()
       at NUnit.Framework.Internal.Execution.WorkItem.RunOnCurrentThread()
       at NUnit.Framework.Internal.Execution.WorkItem.Execute()
       at NUnit.Framework.Internal.Execution.ParallelWorkItemDispatcher.Dispatch(WorkItem work, ParallelExecutionStrategy strategy)
       at NUnit.Framework.Internal.Execution.ParallelWorkItemDispatcher.Dispatch(WorkItem work)
       at NUnit.Framework.Internal.Execution.CompositeWorkItem.RunChildren()
       at NUnit.Framework.Internal.Execution.CompositeWorkItem.PerformWork()
       at NUnit.Framework.Internal.Execution.WorkItem.RunOnCurrentThread()
       at NUnit.Framework.Internal.Execution.WorkItem.Execute()
       at NUnit.Framework.Internal.Execution.TestWorker.TestWorkerThreadProc()
       at System.Threading.Thread.StartHelper.Callback(Object state)
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.Thread.StartCallback()
    2023-06-19T08:39:57+02:00 Error - Config value read (ConfigFile.cs in line 165), A necessary config value could not be found.
     ---
    Stack Trace:
        at System.Environment.get_StackTrace()
       at FWO.Logging.Log.WriteError(String Title, String Text, Boolean LogStackTrace, String callerName, String callerFile, Int32 callerLineNumber) in /usr/local/fworch/lib/files/FWO.Logging/Log.cs:line 138
       at FWO.Config.File.ConfigFile.CriticalConfigValueLoaded[ConfigValueType](ConfigValueType configValue) in /usr/local/fworch/lib/files/FWO.Config.File/ConfigFile.cs:line 165
       at FWO.Config.File.ConfigFile.get_ApiServerUri() in /usr/local/fworch/lib/files/FWO.Config.File/ConfigFile.cs:line 64
       at FWO.Test.ConfigFileTest.<>c.<MissingValueConfigFile>b__12_1() in /usr/local/fworch/test/csharp/FWO.Test/ConfigFileTest.cs:line 130
       at NUnit.Framework.Assert.Throws(IResolveConstraint expression, TestDelegate code, String message, Object[] args)
       at NUnit.Framework.Assert.Throws(IResolveConstraint expression, TestDelegate code)
       at NUnit.Framework.Assert.Catch(Type expectedExceptionType, TestDelegate code)
       at FWO.Test.ConfigFileTest.MissingValueConfigFile() in /usr/local/fworch/test/csharp/FWO.Test/ConfigFileTest.cs:line 130
       at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
       at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
       at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
       at NUnit.Framework.Internal.Reflect.InvokeMethod(MethodInfo method, Object fixture, Object[] args)
       at NUnit.Framework.Internal.MethodWrapper.Invoke(Object fixture, Object[] args)
       at NUnit.Framework.Internal.Commands.TestMethodCommand.InvokeTestMethod(TestExecutionContext context)
       at NUnit.Framework.Internal.Commands.TestMethodCommand.RunTestMethod(TestExecutionContext context)
       at NUnit.Framework.Internal.Commands.TestMethodCommand.Execute(TestExecutionContext context)
       at NUnit.Framework.Internal.Execution.SimpleWorkItem.<>c__DisplayClass4_0.<PerformWork>b__0()
       at NUnit.Framework.Internal.ContextUtils.<>c__DisplayClass1_0`1.<DoIsolated>b__0(Object _)
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
       at NUnit.Framework.Internal.ContextUtils.DoIsolated(ContextCallback callback, Object state)
       at NUnit.Framework.Internal.ContextUtils.DoIsolated[T](Func`1 func)
       at NUnit.Framework.Internal.Execution.SimpleWorkItem.PerformWork()
       at NUnit.Framework.Internal.Execution.WorkItem.RunOnCurrentThread()
       at NUnit.Framework.Internal.Execution.WorkItem.Execute()
       at NUnit.Framework.Internal.Execution.ParallelWorkItemDispatcher.Dispatch(WorkItem work, ParallelExecutionStrategy strategy)
       at NUnit.Framework.Internal.Execution.ParallelWorkItemDispatcher.Dispatch(WorkItem work)
       at NUnit.Framework.Internal.Execution.CompositeWorkItem.RunChildren()
       at NUnit.Framework.Internal.Execution.CompositeWorkItem.PerformWork()
       at NUnit.Framework.Internal.Execution.WorkItem.RunOnCurrentThread()
       at NUnit.Framework.Internal.Execution.WorkItem.Execute()
       at NUnit.Framework.Internal.Execution.TestWorker.TestWorkerThreadProc()
       at System.Threading.Thread.StartHelper.Callback(Object state)
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.Thread.StartCallback()
  
    2023-06-19T08:39:57+02:00 Info - Test Log (HtmlToPdfTest.cs in line 28), starting PDF generation
  
    2023-06-19T08:39:58+02:00 Info - Test Log (ExportTest.cs in line 276), starting changes report html generation
  
    2023-06-19T08:39:58+02:00 Info - Test Log (ExportTest.cs in line 660), starting changes report json generation
  
    2023-06-19T08:39:58+02:00 Info - Test Log (ExportTest.cs in line 228), starting nat rules report html generation
  
    2023-06-19T08:39:58+02:00 Info - Test Log (ExportTest.cs in line 162), starting recert report html generation
  
    2023-06-19T08:39:58+02:00 Info - Test Log (ExportTest.cs in line 499), starting changes report resolved csv generation
  
    2023-06-19T08:39:58+02:00 Info - Test Log (ExportTest.cs in line 339), starting changes report resolved html generation
  
    2023-06-19T08:39:58+02:00 Info - Test Log (ExportTest.cs in line 759), starting resolved changes report json generation
  
    2023-06-19T08:39:58+02:00 Info - Test Log (ExportTest.cs in line 526), starting changes report tech csv generation
  
    2023-06-19T08:39:58+02:00 Info - Test Log (ExportTest.cs in line 398), starting changes report tech html generation
  
    2023-06-19T08:39:58+02:00 Info - Test Log (ExportTest.cs in line 789), starting resolved changes report json generation
  
    2023-06-19T08:39:58+02:00 Info - Test Log (ExportTest.cs in line 459), starting rules report resolved csv generation
  
    2023-06-19T08:39:58+02:00 Info - Test Log (ExportTest.cs in line 94), starting rules report resolved html generation
  
    2023-06-19T08:39:58+02:00 Info - Test Log (ExportTest.cs in line 613), starting resolved rules report json generation
  
    2023-06-19T08:39:58+02:00 Info - Test Log (ExportTest.cs in line 479), starting rules report tech csv generation
  
    2023-06-19T08:39:58+02:00 Info - Test Log (ExportTest.cs in line 128), starting rules report resolved html generation
  
    2023-06-19T08:39:58+02:00 Info - Test Log (ExportTest.cs in line 636), starting resolved rules report tech json generation
  
    2023-06-19T08:39:58+02:00 Info - Test Log (ExportTest.cs in line 45), starting rules report html generation
  
    2023-06-19T08:39:58+02:00 Info - Test Log (ExportTest.cs in line 548), starting rules report json generation
  
  
    Failed!  - Failed:     1, Passed:    42, Skipped:     0, Total:    43, Duration: 740 ms - /usr/local/fworch/test/csharp/FWO.Test/bin/Debug/net6.0/FWO.Test.dll (net6.0)
  stdout_lines: <omitted>

RUNNING HANDLER [test : delete test user and dir] ***********************************************************************************************************************************************************************
ok: [localhost]

```

--- a/roles/middleware/files/FWO.Middleware.Server/Ldap.cs
+++ b/roles/middleware/files/FWO.Middleware.Server/Ldap.cs
@@ -184,9 +184,13 @@ namespace FWO.Middleware.Server
                     }
                 }
             }
+            catch (LdapException ldapException)
+            {
+                Log.WriteInfo("Ldap entry excpetion", $"Ldap entry search at \"{Address}:{Port}\" lead to exception: {ldapException.Message}");
+            }
             catch (Exception exception)
             {
-                Log.WriteError($"Non-LDAP exception {Address}:{Port}", "Unexpected error while trying to validate user", exception);
+                Log.WriteError($"Non-LDAP exception \"{Address}:{Port}\"", "Unexpected error while trying to validate user", exception);
             }
 
             Log.WriteDebug("Invalid Credentials", $"Invalid login credentials - could not authenticate user \"{ user.Name}\" on {Address}:{Port}.");

--- a/roles/middleware/files/FWO.Middleware.Server/Ldap.cs
+++ b/roles/middleware/files/FWO.Middleware.Server/Ldap.cs
@@ -186,7 +186,7 @@ namespace FWO.Middleware.Server
             }
             catch (LdapException ldapException)
             {
-                Log.WriteInfo("Ldap entry excpetion", $"Ldap entry search at \"{Address}:{Port}\" lead to exception: {ldapException.Message}");
+                Log.WriteInfo("Ldap entry exception", $"Ldap entry search at \"{Address}:{Port}\" lead to exception: {ldapException.Message}");
             }
             catch (Exception exception)
             {

--- a/roles/test/files/FWO.Test/LockTest.cs
+++ b/roles/test/files/FWO.Test/LockTest.cs
@@ -1,5 +1,6 @@
 ﻿using FWO.Logging;
 using NUnit.Framework;
+using System;
 using System.Reflection;
 
 namespace FWO.Test
@@ -9,26 +10,35 @@ namespace FWO.Test
     public class LockTest
     {
         private string lockFilePath = $"/var/fworch/lock/{Assembly.GetEntryAssembly()?.GetName().Name}_log.lock";
+        private static Random random = new Random();
 
         [SetUp]
-        public void SetUp()
+        public async Task SetUp()
         {
-            if (File.Exists(lockFilePath))
-            {
-                File.Delete(lockFilePath);
-            }
+			await ExecuteFileAction(() =>
+			{
+				if (File.Exists(lockFilePath))
+				{
+					File.Delete(lockFilePath);
+				}
+				return Task.CompletedTask;
+			});
 
             // Implicitly call static constructor so backround lock process is started
             Log.WriteInfo("Startup", "Starting Lock Tests...");
         }
 
         [TearDown]
-        public void TearDown()
+        public async Task TearDown()
         {
-            if (File.Exists(lockFilePath))
-            {
-                File.Delete(lockFilePath);
-            }
+			await ExecuteFileAction(() =>
+			{
+				if (File.Exists(lockFilePath))
+				{
+					File.Delete(lockFilePath);
+				}
+				return Task.CompletedTask;
+			});
         }
 
         [Test]
@@ -36,53 +46,95 @@ namespace FWO.Test
         public async Task LogLockUi()
         {
             // Request lock
-            using (var writer = new StreamWriter(lockFilePath))
+            await ExecuteFileAction(async () =>
             {
-                await writer.WriteLineAsync("REQUESTED");
-            }
+                using (var writer = new StreamWriter(lockFilePath))
+                {
+                    await writer.WriteLineAsync("REQUESTED");
+                }
+            });
 
-            await Task.Delay(1200);
+			await Task.Delay(1200);
 
-            // Assure lock is granted after request
-            using (var reader = new StreamReader(lockFilePath))
-            {
-                Assert.That((await reader.ReadToEndAsync()).Trim().EndsWith("GRANTED"));
-            }
+			// Assure lock is granted after request
+			await ExecuteFileAction(async () =>
+			{
+				using (var reader = new StreamReader(lockFilePath))
+				{
+					Assert.That((await reader.ReadToEndAsync()).Trim().EndsWith("GRANTED"));
+				}
+			});
 
             // Assure write is NOT possible after lock was granted
             Task logWriter = Task.Run(() =>
             {
-                Log.WriteDebug("TEST_TILE", "TEST_TEXT");
+                Log.WriteDebug("TEST_TITLE", "TEST_TEXT");
             });
 
             await Task.Delay(500);
 
             Assert.That(logWriter.IsCompleted, Is.False);
 
-            // Release lock
-            using (var writer = new StreamWriter(lockFilePath))
-            {
-                await writer.WriteLineAsync("RELEASED");
-            }
+			// Release lock
+			await ExecuteFileAction(async () =>
+			{
+				using (var writer = new StreamWriter(lockFilePath))
+				{
+					await writer.WriteLineAsync("RELEASED");
+				}
+			});
 
             await Task.Delay(1200);
 
             // Assure write IS possible after lock was released
             Assert.That(logWriter.IsCompleted, Is.True);
 
-            // Request lock
-            using (var writer = new StreamWriter(lockFilePath))
-            {
-                await writer.WriteLineAsync("REQUESTED");
-            }
+			// Request lock
+			await ExecuteFileAction(async () =>
+			{
+				using (var writer = new StreamWriter(lockFilePath))
+				{
+					await writer.WriteLineAsync("REQUESTED");
+				}
+			});
 
             await Task.Delay(11_200);
 
-            // If not release in time make sure that the lock will be forcefully released
-            using (var reader = new StreamReader(lockFilePath))
-            {
-                Assert.That((await reader.ReadToEndAsync()).Trim().EndsWith("FORCEFULLY RELEASED"));
-            }
+			// If not release in time make sure that the lock will be forcefully released
+			await ExecuteFileAction(async () =>
+			{
+				using (var reader = new StreamReader(lockFilePath))
+				{
+					Assert.That((await reader.ReadToEndAsync()).Trim().EndsWith("FORCEFULLY RELEASED"));
+				}
+			});
         }
-    }
+
+		private static async Task ExecuteFileAction(Func<Task> action)
+		{
+			bool success = false;
+			int maxRetryAttempts = 50;
+			int retryCount = 0;
+
+			// Handle IO Exception like file blocking from another process by retrying with a random delay
+			while (!success && retryCount < maxRetryAttempts)
+			{
+				try
+				{
+                    await action();
+					success = true;
+				}
+				catch (IOException)
+				{
+					retryCount++;
+				}
+				await Task.Delay(random.Next(50, 100));
+			}
+
+			if (!success)
+			{
+                Assert.Fail($"Lock file access failed after {maxRetryAttempts} retries.");
+			}
+		}
+	}
 }

--- a/roles/ui/files/FWO.UI/Pages/Settings/SettingsGateways.razor
+++ b/roles/ui/files/FWO.UI/Pages/Settings/SettingsGateways.razor
@@ -231,6 +231,7 @@
     private void Edit(Device device)
     {
         actDevice = new Device(device);
+        setMatchingDevTypes(actDevice.Management);
         EditMode = true;
     }
 

